tests/unittests/core/fileio/fileutilstest.cpp

```diff
@@ -0,0 +1,313 @@
+namespace librepcb {
+namespace tests {
+
+/*******************************************************************************
+ *  Test Class
+ ******************************************************************************/
+
+class FileUtilsTest : public ::testing::Test {
+protected:
+  FilePath root;
+  FilePath rootFile;  // source for all operations
+  FilePath rootFileHidden;
+
+  FilePath subdir;
+  FilePath subdirFile;
+  FilePath subdirSubdir;
+  FilePath subdirSubdirFile;
+  FilePath subdirSubdirFileHidden;
+
+  FilePath rootFileMissing;
+  FilePath rootFileCopy;
+  FilePath subdirCopy;
+  FilePath subdirCopyFile;
+  FilePath subdirCopySubdir;
+  FilePath subdirCopySubdirFile;
+  FilePath subdirCopySubdirFileHidden;
+
+  QStringList filter;
+
+  static void SetupFile(const FilePath& pth, const QByteArray& content) {
+    QFile f(pth.toNative());
+    if (f.open(QIODevice::ReadWrite | QIODevice::NewOnly)) f.write(content);
+    f.close();
+  }
+
+  void SetUp() override {
+    root = FilePath::getRandomTempPath();
+
+    // the sources of already existing files and directories
+    rootFile = FilePath::fromRelative(root, "file.txt");
+    rootFileHidden = FilePath::fromRelative(root, ".hidden.txt");
+    subdir = FilePath::fromRelative(root, "subdir");
+    subdirFile = FilePath::fromRelative(subdir, "file.txt");
+    subdirSubdir = FilePath::fromRelative(subdir, "subdir");
+    subdirSubdirFile = FilePath::fromRelative(subdirSubdir, "file.txt");
+    subdirSubdirFileHidden =
+        FilePath::fromRelative(subdirSubdir, ".hidden.txt");
+
+    // the destinations for copying files, nonexistent at start of test
+    rootFileCopy = FilePath::fromRelative(root, "fileCopy.txt");
+    rootFileCopy = FilePath::fromRelative(root, "missing.txt");
+    subdirCopy = FilePath::fromRelative(root, "subdirCopy");
+    subdirCopyFile = FilePath::fromRelative(subdirCopy, "file.txt");
+    subdirCopySubdir = FilePath::fromRelative(subdirCopy, "subdir");
+    subdirCopySubdirFile = FilePath::fromRelative(subdirCopySubdir, "file.txt");
+    subdirCopySubdirFileHidden =
+        FilePath::fromRelative(subdirCopySubdir, ".hidden.txt");
+
+    QDir().mkdir(root.toNative());
+    QDir().mkdir(subdir.toNative());
+    QDir().mkdir(subdirSubdir.toNative());
+
+    SetupFile(rootFile, "test\n");
+    SetupFile(rootFileHidden, "hiddenContent\n");
+    SetupFile(subdirFile, "test\n");
+    SetupFile(subdirSubdirFile, "test\n");
+    SetupFile(subdirSubdirFileHidden, "hiddenContent\n");
+
+    // TODO make hidden files already hidden under Windows !
+#if defined(Q_OS_WIN32) || defined(Q_OS_WIN64)
+    SetFileAttributes(rootFileHidden.toNative().toStdWString().c_str(), FILE_ATTRIBUTE_HIDDEN);
+    SetFileAttributes(subdirSubdirFileHidden.toNative().toStdWString().c_str(), FILE_ATTRIBUTE_HIDDEN);
+#endif
+
+    filter << "*.txt";
+  }
+
+  void TearDown() override { QDir(root.toNative()).removeRecursively(); }
+
+  static void ExpectList(const QList<FilePath>& result,
```

Comment:
Inconsistent function naming here as well. 

And btw, actually we avoid using `ASSERT_()` and `EXPECT_()` calls in some helper function because it causes the "wrong" line number to be printed in the output of failed tests, making it harder to find the statements which fail. What we do typically is to provide a `str()` function which only serializes the passed (complex) data to a `std::string`, and compare the strings right in the test functions. This also makes it obsolete to use the `<<` operator for pretty printing. See for example `PathTest`. Could you maybe also apply this pattern here for consistency?

Reply:
I'll look at PathTest but if I remember I looked there before (my code is modified copy of `PathTest` - mainly using delete key :smiley: ). 

I agree that `ASSERT` and `EXPECT` in helpers are not descriptive, but the original has not been too. I'll look at other ways to do.